result/src/main/kotlin/com/github/kittinunf/result/Result.kt

```diff
@@ -4,59 +4,96 @@ package com.github.kittinunf.result
  * Created by Kittinun Vantasin on 10/26/15.
  */
 
-public interface ResultType<out V : Any, out E : Exception> {
```

Comment:
Ok noted. Yeah, actually I designed it as I feel it offers more flexibility. However, I don't know whether I would run into those case just yet. Maybe you are right that just a simple `Result` should be enough.